27). Add topic files 79 to 82 (#18081)
dotnet/docs: docs/standard/linq/find-root-element.md

---
title: How to find the root element - LINQ to XML

> description: This article provides an example that shows how to use XPath and LINQ to XML, in C# and Visual Basic, to find the root element of an XML document.

Reviewer: ```suggestion
description: Learn how to use XPath and LINQ to XML, in C# and Visual Basic, to find the root element of an XML document.
```

  - "vb"
ms.assetid: 4fd824e0-4d39-429b-b092-f6a5c046ee6c
---

# How to find the root element (LINQ to XML)

This article provides an example that shows how to use XPath and LINQ to XML, in C# and Visual Basic, to find the root element of an XML document.

## Example: Find the root element

This example uses LINQ to XML query and XPath to find the root element in XML document [Sample XML file: Multiple purchase orders](sample-xml-file-multiple-purchase-orders.md). The XPath expression is `/PurchaseOrders`.

```csharp
XDocument po = XDocument.Load("PurchaseOrders.xml");

// LINQ to XML query
XElement el1 = po.Root;

// XPath expression
XElement el2 = po.XPathSelectElement("/PurchaseOrders");

if (el1 == el2)
    Console.WriteLine("Results are identical");
else
    Console.WriteLine("Results differ");
Console.WriteLine(el1.Name);
```

```vb
Dim po As XDocument = XDocument.Load("PurchaseOrders.xml")

' LINQ to XML query
Dim el1 As XElement = po.Root

' XPath expression
Dim el2 As XElement = po.XPathSelectElement("/PurchaseOrders")

If el1 Is el2 Then
    Console.WriteLine("Results are identical")
Else
    Console.WriteLine("Results differ")
End If
Console.WriteLine(el1.Name)
```

> The example produces this output:

Reviewer: ```suggestion
The example produces the following output:
```
